install.sh

```diff
@@ -0,0 +1,152 @@
+https://www.shuttle.rs
+https://github.com/shuttle-hq/shuttle
+
+Please file an issue if you encounter any problems!
+===================================================
+EOF
+
+if ! command -v curl &>/dev/null; then
+  echo "curl not installed. Please install curl."
+  exit
+fi
+
+REPO_URL="https://github.com/shuttle-hq/shuttle"
+LATEST_RELEASE=$(curl -L -s -H 'Accept: application/json' "$REPO_URL/releases/latest")
+LATEST_VERSION="${LATEST_RELEASE#*\"tag_name\":\"}"
+LATEST_VERSION="${LATEST_VERSION%%\"*}"
+
+_install_linux() {
+  echo "Detected Linux!"
+  echo "Checking distro..."
+  if (uname -a | grep -qi "Microsoft"); then
+    OS="ubuntuwsl"
+  elif ! command -v lsb_release &>/dev/null; then
+    echo "lsb_release could not be found. Falling back to /etc/os-release"
+    OS="$(grep -Po '(?<=^ID=).*$' /etc/os-release | tr '[:upper:]' '[:lower:]')" 2>/dev/null
+  else
+    OS=$(lsb_release -i | awk '{ print $3 }' | tr '[:upper:]' '[:lower:]')
+  fi
+  case "$OS" in
+  "arch" | "manjarolinux" | "endeavouros")
+    _install_arch_linux
+    ;;
+  "ubuntu" | "ubuntuwsl" | "debian" | "linuxmint" | "parrot" | "kali" | "elementary" | "pop")
+    # TODO: distribute .deb packages via `cargo-deb` and install them here
+    _install_unsupported
+    ;;
+  *)
+    _install_unsupported
+    ;;
+  esac
+}
+
+_install_arch_linux() {
+  echo "Arch Linux detected!"
+  if command -v pacman &>/dev/null; then
+    echo "Installing with pacman"
+    sudo pacman -S cargo-shuttle
+  else
+    echo "Pacman not found"
+    exit 1
+  fi
+}
+
+# TODO: package cargo-shuttle for Homebrew
+_install_mac() {
+  _install_unsupported
+}
+
+_install_binary() {
+  echo "Installing pre-built binary..."
+  case "$OSTYPE" in
+  linux*) target="x86_64-unknown-linux-musl" ;;
+  darwin*) target="x86_64-apple-darwin" ;;
+  *)
+    echo "Cannot determine the target to install"
+    exit 1
+    ;;
+  esac
+  temp_dir=$(mktemp -d)
+  pushd "$temp_dir" >/dev/null || exit 1
+  curl -LO "$REPO_URL/releases/download/$LATEST_VERSION/cargo-shuttle-$LATEST_VERSION-$target.tar.gz"
+  tar -xzf "cargo-shuttle-$LATEST_VERSION-$target.tar.gz"
+  echo "Installing to $HOME/.cargo/bin/cargo-shuttle"
+  mv "cargo-shuttle-$target-$LATEST_VERSION/cargo-shuttle" "$HOME/.cargo/bin/"
+  popd >/dev/null || exit 1
+  if [[ ":$PATH:" != *":$HOME/.cargo/bin:"* ]]; then
+    echo "Add $HOME/.cargo/bin to PATH to run cargo-shuttle"
+  fi
+}
+
+_install_cargo() {
+  echo "Attempting install with cargo"
+  if ! command -v cargo &>/dev/null; then
+    echo "cargo not found! Attempting to install rustup"
+    if command -v rustup &>/dev/null; then
+      echo "rustup was found, but cargo wasn't. Something is up with your install"
+      exit 1
+    fi
+    curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -q
+    echo "rustup installed! Attempting cargo install"
+  fi
+
+  cargo install cargo-shuttle
+}
+
+_install_unsupported() {
+  echo "Unknown or unsupported OS or architecture"
```

Comment:
I wouldn't call these platforms unsupported... maybe "installing with package manager not supported".
Btw, going via package manager might get you an outdated version, so this route is the "most" reliable.

Reply:
Imagine on a new release: "Please run the install script to upgrade!" -> package manager not updated -> wow

Reply:
> Btw, going via package manager might get you an outdated version, so this route is the "most" reliable.

Although it is true that you might get an outdated version, I disagree with the reliability statement. Using package managers are usually more reliable than installing pre-built binaries since the packages are built specifically for that distro (and sometimes even have necessary fix/security patches for that package to work).

> maybe "installing with package manager not supported".

eff50729efb869548708e96712804395e349896c

Reply:
Agree with your point, but since we are lowkey requiring the latest version to be installed (for now), the scenario "don't work cause outdated" feels more likely than "don't work cause not installed properly with package manager" in Shuttle's case.
Maybe we can introduce a check+prompt "latest version not yet available through <package manager>, install with binary/cargo instead?". My main concern is
> Imagine on a new release: "Please run the install script to upgrade!" -> package manager not updated -> wow

Reply:
We are adding a warning if pacman does not have latest version yet.